view/graph/GraphArea.kt

```diff
@@ -124,62 +122,58 @@ class GraphArea(transactionState: TransactionState) {
 
     private fun rendererContext(drawScope: DrawScope) = RendererContext(drawScope, theme!!)
 
-    fun vertexBackgroundRenderer(vertex: Vertex, drawScope: DrawScope): VertexBackgroundRenderer {
+    private fun vertexBackgroundRenderer(vertex: Vertex, drawScope: DrawScope): VertexBackgroundRenderer {
         return VertexBackgroundRenderer.of(vertex, this, rendererContext(drawScope))
     }
 
-    fun edgeRenderer(drawScope: DrawScope): EdgeRenderer {
+    private fun edgeRenderer(drawScope: DrawScope): EdgeRenderer {
         return EdgeRenderer(this, rendererContext(drawScope))
     }
 
     @Composable
     private fun EdgeLayer() {
```

Comment:
### Edge rendering

For edges, we have two draw modes: _simple_ and _detailed_. A _detailed_ edge has a label, and also, detailed edges can be curved. Because `DrawScope.drawPoints` is so cheap, we can draw all edges as simple by default, and detailed when they meet certain criteria:

- Draw edges as detailed if the label is in the viewport, AND viewport scale is > 0.2, AND (physics alpha < 0.25 OR total no. of detailed edges is < 100). This ensures smooth performance when zoomed out and during the initial explosion.
- As an additional optimisation, if viewport scale < 0.2 AND total edges in graph > 500, ALL edges are drawn in simple mode. This ensures smooth performance when zoomed out, and in this case we don't need to determine which edge labels are visible, saving resources.